client/my-sites/site-settings/header-code/index.jsx

```diff
@@ -0,0 +1,182 @@
+	isSavingSettings,
+	saveSettings,
+	siteId,
+	siteSlug,
+	successMessage,
+	translate,
+} ) => {
+	const [ htmlCode, setHtmlCode ] = useState( '' );
+
+	useEffect( () => {
+		if ( 'string' === typeof headCode && headCode.length ) {
+			setHtmlCode( headCode );
+		}
+
+		// Necessary for if switching to a site without any code.
+		if ( ! headCode ) {
+			setHtmlCode( '' );
+		}
+	}, [ headCode, siteId ] );
+
+	useEffect( () => {
+		if ( isSaveSuccess ) {
+			successMessage( translate( 'Code updated successfully!' ), {
+				duration: 4000,
+			} );
+		} else if ( isSaveFailure && ! isSavingSettings ) {
+			errorMessage( translate( 'There was an error saving your code.' ), {
+				duration: 4000,
+			} );
+		}
+	}, [ isSaveSuccess, isSavingSettings, isSaveFailure ] );
+
+	const handleSaveSettings = () => {
+		saveSettings( siteId, {
+			jetpack_header_code: htmlCode,
+		} );
+	};
+
+	const isSupported = isJetpack || hasPlanFeature;
+
+	return (
+		<Main className="header-code">
+			<DocumentHead title={ translate( 'Header Code' ) } />
+			{ siteId && <QuerySiteSettings siteId={ siteId } /> }
+			{ ! isAdmin && (
+				<EmptyContent
+					illustration={ illustration404 }
+					title={ translate( 'You are not authorized to view this page' ) }
+				/>
+			) }
+
+			{ isAdmin && (
+				<>
+					<NavigationHeader
+						title={ translate( 'Header Code' ) }
+						subtitle={ translate( 'Insert code within the Header element of your site.' ) }
+					/>
+					<HeaderCake backHref={ `/settings/general/${ siteSlug }` } isCompact>
+						<h1>{ translate( 'Header Code' ) }</h1>
+					</HeaderCake>
+					<Card>
+						<UpsellNudge
+							plan={ PLAN_BUSINESS }
+							title={ translate( 'Add custom code to your site with the %(planName)s plan.', {
+								args: { planName: getPlan( PLAN_BUSINESS ).getTitle() },
+							} ) }
+							description={ translate(
+								'Upgrade to insert JavaScript and HTML meta tags to your site.'
+							) }
+							showIcon
+							forceDisplay={ ! isSupported }
+						/>
+						<p className="header-code__text">
+							{ translate(
+								'The following code will be inserted within {{code}}<head>{{/code}} on your site. {{link}}Learn more{{/link}}.',
+								{
+									components: {
+										code: <code />,
+										link: (
+											<InlineSupportLink supportContext="insert-header-code" showIcon={ false } />
+										),
+									},
+								}
+							) }
+						</p>
+						<div
+							className={ classNames( 'header-code__editor', {
+								'is-loading': isRequestingSettings,
+							} ) }
+						>
+							<CodeMirror
+								value={ htmlCode }
+								onChange={ ( value ) => setHtmlCode( value ) }
+								height="235px"
+								extensions={ [ html() ] }
+							/>
+						</div>
+						<div className="header-code__warning">
+							<Gridicon icon="info" size={ 18 } />
+							<p className="header-code__warning-text">
+								{ translate( "This code will be run for all your site's viewers." ) }{ ' ' }
```

Comment:
Leaving spaces like this is awkward. We should either use a single string, or semantically separate these into 2 pieces.